runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -0,0 +1,239 @@
+local util = require('vim.lsp.util')
+local log = require('vim.lsp.log')
+local handlers = require('vim.lsp.handlers')
+local api = vim.api
+local M = {}
+
+-- [[
+-- TODO:
+-- [ ] only request for visible range
+-- ]]
+--
+---@class lsp.inlay_hint.bufstate
+---@field version integer
+---@field client_hint table<integer, table<integer, lsp.InlayHint[]>> client_id -> (lnum -> hints)
+
+---@type table<integer, lsp.inlay_hint.bufstate>
+local hint_cache_by_buf = setmetatable({}, {
+  __index = function(t, b)
+    local key = b > 0 and b or api.nvim_get_current_buf()
+    return rawget(t, key)
+  end,
+})
+
+--- Store hints for a specific buffer and client
+--- Resolves unresolved hints
+---
+---@param hints lsp.InlayHint[]|nil (table) of lenses to store
+---@param bufnr integer
+---@param client_id integer
+---@param version integer
+function M.save(hints, bufnr, client_id, version)
```

Comment:
```suggestion
function M.set(hints, bufnr, client_id, version)
```

See `:h dev-naming`. I know this was taken from codelens, but the name unfortunately doesn't match usual naming conventions.

Or maybe even remove this and inline it into `on_inlayhint`. Not sure if we need two APIs to set the hints